openid-connect-server/src/test/java/org/mitre/oauth2/service/impl/TestDefaultOAuth2AuthorizationCodeService

```diff
@@ -0,0 +1,129 @@
+import org.mockito.InjectMocks;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.common.util.RandomValueStringGenerator;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestDefaultOAuth2AuthorizationCodeService {
+	
+	
+	private String codeNotFound;
+	
+	private AuthenticationHolderEntity authHolder;
+	
+	private AuthorizationCodeEntity entity;
+	
+	//are these 3 mocks important?
+	@Mock
+	private AuthorizationCodeRepository repository;
+	
+	@Mock
+	private AuthenticationHolderRepository authenticationHolderRepository;
+	
+	
+	//following two mock suggested by justin
+	@Mock
+	private JpaAuthenticationHolderRepository  authRepo;
+	
+	@Mock
+	private OAuth2Authentication authentication;
+	
+	
+
+	@InjectMocks
+	private DefaultOAuth2AuthorizationCodeService service;
+	
+	
+
+	@Before
+	public void setUp() {
+		
+		Mockito.reset(repository);
+		Mockito.reset(authenticationHolderRepository);
+		codeNotFound = "codeNotFound";
+		Mockito.when(repository.getByCode(codeNotFound)).thenReturn(null);
+		authHolder = new AuthenticationHolderEntity();
+		authHolder.setAuthentication(authentication);
+		authenticationHolderRepository.save(authHolder);
+		entity = new AuthorizationCodeEntity();
+
+	}
+	
+	@Test
+	public void testCreateAuthorizationCode(){
+		
+		OAuth2Request auth2Request = Mockito.mock(OAuth2Request.class);
+		Authentication authObj = Mockito.mock(Authentication.class);
+		OAuth2Authentication authentication = new OAuth2Authentication(auth2Request, authObj);
+		String code = service.createAuthorizationCode(authentication);
+		ArgumentCaptor<AuthorizationCodeEntity> argumentCaptor = ArgumentCaptor.forClass(AuthorizationCodeEntity.class);
+		Mockito.verify(repository).save(argumentCaptor.capture());
+		assertThat(argumentCaptor.getValue(), equalTo(code));
+	
+		}
+
+	@Test
+	public void testConsumeAuthorizationCodeFound() {
+		
+		String code = service.createAuthorizationCode(authentication);
+		assertEquals(repository.getByCode(code),authHolder);
+		when(repository.save(Matchers.isA(AuthorizationCodeEntity.class))).thenReturn(entity);
+		
+	}
+
+	@Test
+	public void testConsumeAuthorizationCodeNotFound() {
+		assertThat(null, equalTo(service.consumeAuthorizationCode(codeNotFound)));	
+	}
+	
+
+	@Test
+	public void testConsumeAuthorizationCode(){
+
+		doNothing().when(authHolder).setAuthentication(Matchers.isA(OAuth2Authentication.class));
+		when(authRepo.save(Matchers.isA(AuthenticationHolderEntity.class))).thenReturn(authHolder);
+		}
+	
+	@Test
+	public void testRespositoryGetAndSet() {
+		service.setRepository(repository);
+		assertEquals(service.getRepository(), repository);			
+	}
+	
+	@Test
+	public void testExpirationSecond() {
```

Comment:
Add another test that checks whether the authorization code that is saved is set with the appropriate seconds. So something like this horrible pseudocode: 

service.setAuthCodeExpiration(99)
verify(respository.save(argumentcaptor))
assertThat(argumentcaptor.expiration().isEqualTo(Date.now() + 99seconds)

Reply:
Thanks Justin for the pseudo.
Will you please tell me what is .expiration() here.

Reply:
It's the expiration of the authorization code that you're testing, you need to make sure that it expires off in the future by the correct amount.

Reply:
Thanks, Justin.
I have made a new test as per your suggestion for expiration code. kindly go through it my recent commit.